### PR TITLE
Update rust-channel to nightly-2020-04-19

### PR DESCRIPTION
**Pull request by Ma27**
Also needed another patch for `wee_alloc` (https://github.com/Ma27/wee_alloc/commit/0f9954eddcbd210a296333f6106c8c492d78ea12).

Todos before merging:

* [ ] consider creating a PR for `wee_alloc` to support latest nightly
* [ ] merging the `wee_alloc` patch into our fork (and give me push-access :p)
* [ ] tests/reviews to make sure my changes to the allocator API are at least somehow reasonable

### CODE REVIEW

**Comment by TrueDoctor** [comment]
• You should have push access now
• can you trigger the build of a new docker container? (just uptate the Dockerfile and push to master) and then update the drone.yaml to pass ci tests

File: client/shared/src/alloc.rs [hunk 1]
```diff
@@ -101,7 +101,7 @@ unsafe impl<M: MutableAlloc + Sized + 'static, S: AllocSettings, I: Initial<M>>
     for Allocator<M, S, I>
 {
     unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
-        let out = Self::allocator().alloc(layout);
+        let out = Self::allocator().alloc(layout, AllocInit::Uninitialized);
```
Comment (TrueDoctor) [inline client/shared/src/alloc.rs:104]: Should be zeroed because that is guaranteed in wasm

**Comment by TrueDoctor** [comment]
Investigate possible usage of `init_offset` to avoid evil hacks in our code

**Comment by TrueDoctor** [comment]
deprecate the simple alloc and tidy up code @JanPlusPlus

**Comment by Ma27** [comment]
> You should have push access now

Thanks!

>  • can you trigger the build of a new docker container? (just uptate the Dockerfile and push to master) and then update the drone.yaml to pass ci tests

Done. I assume we have to restart the build once again when the container is built?

**Comment by TrueDoctor** [comment]
> Done. I assume we have to restart the build once again when the container is built?

Sorry, I should have mentioned, that you have to change the docker tag in the drone.yaml as well.
Now the new version is published under the old tag

**Comment by TrueDoctor** [comment]
Do you know why the container still downloads dependencies during the `test`/`format` steps?
This does not occur during the `build` step and leads to longer ci times (the ci should not take more than 5 minutes for one build, currently 6:30 min)

**Comment by Ma27** [comment]
@TrueDoctor I updated the nixpkgs rev in this change, but I decided to keep building the docker container using the `shell.nix` from master. Due to the newer revision, several things changed and we therefore need to download more stuff. By either merging this PR or building the container with the shell.nix from this branch, that should be fixed I guess.

**Pull request merged**